src/Storages/StorageFuzzJSON.cpp

```diff
@@ -0,0 +1,511 @@
+#include <optional>
+#include <random>
+#include <Columns/ColumnString.h>
+#include <Storages/StorageFactory.h>
+#include <Storages/StorageFuzzJSON.h>
+#include <Storages/checkAndGetLiteralArgument.h>
+#include <Common/JSONParsers/RapidJSONParser.h>
+#include <Common/JSONParsers/SimdJSONParser.h>
+#include <Common/checkStackSize.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+
+extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+extern const int NOT_IMPLEMENTED;
+extern const int INCORRECT_DATA;
+}
+
+using uniform = std::uniform_int_distribution<size_t>;
+using Node = StorageFuzzJSON::JSONNode;
+using JSONNodeList = std::list<std::shared_ptr<Node>>;
+
+using FuzzConfig = StorageFuzzJSON::Config;
+
+struct StorageFuzzJSON::JSONNode
```

Comment:
Can't we use some existing JSON node implementation, such as `Poco::JSON::Object`? Will it make code more comlicated or less performant?

Reply:
Thank you for the review! Here's what I think about using `Poco::JSON::Object`. I'll respond to the remaining comments tomorrow.

Indeed, Poco::JSON::Object [stores data](https://github.com/ClickHouse/ClickHouse/blob/master/base/poco/JSON/include/Poco/JSON/Object.h#L290-L297) that is not needed for the task, but that is not my primary concern.
  
## 
If we choose Poco, a node will essentially be a wildcard of the `Poco::Dynamic::Var` type, as per the implementation Poco::JSON::Object. That can store any type of data, and the parsing (and fuzzing) will depend upon `typeid` checks:

https://github.com/ClickHouse/ClickHouse/blob/44bd0598a3c128646974bc114dac2c44797159d7/base/poco/JSON/include/Poco/JSON/Object.h#L357

https://github.com/ClickHouse/ClickHouse/blob/44bd0598a3c128646974bc114dac2c44797159d7/base/poco/JSON/include/Poco/JSON/Object.h#L377

I've played with Poco a bit, and here is how the traversal (and fuzzing) of a JSON will look like (more or less):
``` c++
using Var = Poco::Dynamic::Var;
using Object = Poco::JSON::Object;
using Array = Poco::JSON::Array;

void printPocoJSON(const Var& var, WriteBuffer & out) {
    if (var.type() == typeid(Array::Ptr)) {
        out << "[";
        bool first = true;
        auto ptr = var.extract<Array::Ptr>();
        for (const auto& elem : *ptr) {
            if (!first)
                out << ", ";
            first = false;

            printPocoJSON(elem, out);
        }
        out << "]";
    } else if (var.type() == typeid(Object::Ptr)) {
        auto ptr = var.extract<Object::Ptr>();
        out << "{";
        bool first = true;
        for (const auto& [k, v] : *ptr) {
            if (!first)
                out << ", ";
            first = false;

            out << k << ":";
            printPocoJSON(v, out);
        }
        out << "}";
    } else {
        out << var.toString();
    }
}
```

What I don't like in that code is the direct access to a variant, followed by the use of typecasts.

`Poco::JSON::Object` does not provide the necessary interface for accessing the underlying elements without exposing the `Poco::Dynamic::Var` type, e.g.:

https://github.com/ClickHouse/ClickHouse/blob/44bd0598a3c128646974bc114dac2c44797159d7/base/poco/JSON/include/Poco/JSON/Object.h#L413
 
I believe that variant types can be dangerous and, if used, the `typeid` casts should be hidden within the internals of the lib and should not be handed off to a user.

## 
Another concern is that the iterator does not preserve the original key order.
Yes, `Object` has a `_preserveInsOrder` flag, yet I don't see that it is of any use.

Given the query
```
 SELECT * FROM fuzzJSON('{"name":"Alice", "hobbies": ["eat", {"drink":["wine", "mojito"]}, {"dance":"tango"} ], "address":{"city":"Plesantview"}}', NULL, 300) LIMIT 10;
```
In the output of the code  snippet above:
```
{address:{city:Plesantview}, hobbies:[eat, {drink:[wine, mojito]}, {dance:tango}], name:Alice}
```